test/e2e/imgpkg_bundle_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_FetchAndDeployImgpkgBundle_Successfully(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, logger}
+	sas := ServiceAccounts{env.Namespace}
+
+	appYaml := fmt.Sprintf(`
+---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: test-bundle-app
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+    - imgpkgBundle:
+        image: k8slt/kappctrl-e2e-bundle
```

Comment:
where are the contents of this bundle? i think we should at least check them into this repo under test/e2e/assets/___.

Reply:
Sure, it's just the example imgpkg bundle: https://github.com/vmware-tanzu/carvel-imgpkg/tree/11c34e5875f29ce7bcbfeb203871ada65bdb60cc/examples/basic-step-2

Will add to assets folder.

Reply:
Added.